src/lib.rs

```diff
@@ -196,39 +196,24 @@ mod tests {
                             self.registers[index_x] ^= self.registers[index_y];
                         },
                         0x0004 => {
-                            let temp : u16 = (self.registers[index_x] + self.registers[index_y]) as u16;
-                            self.registers[index_x] += self.registers[index_y];
-                            if temp & 0xFF00 != 0 {
-                                self.registers[0xF] = 1;
-                            }
-                            else {
-                                self.registers[0xF] = 0;
-                            }
+                            let (vx, vf) = self.registers[index_x].overflowing_add(self.registers[index_y]);
+                            self.registers[index_x] = vx;
+                            self.registers[0xF] = vf as u8;
```

Comment:
just confirming, does this cast take a bool to a u8? Since I am unfamiliar with rust can you confirm that bool true cast to u8 is 1 and false is 0?

Reply:
[The docs](https://doc.rust-lang.org/std/primitive.bool.html) specify that bool as int will be either 1 or 0 respectably. I also went ahead a double checked using the following snippet:
```
let test = true;
assert_eq!(test as u8, 1);
```

Reply:
sweet. wasn't sure since you can't do something like
```
let numeric_value : u8 = 8;
if numeric_value {
    // Something
}
```